analyzer/codechecker_analyzer/analyzers/clangsa/ctu_manager.py

```diff
@@ -134,20 +138,33 @@ def generate_ast(triple_arch, action, source, config, env):
                   cmdstr, err)
 
 
-def ast_dump_path(source_path):
+def ast_dump_path(path, config, triple_arch):
     """ AST-dump based analysis uses preprocessed paths, here the path prefix
     'ast' and the filename suffix '.ast' is added to the name of the original
     source file. """
 
+    # If clang-extdef-mapping can read a pch then the file entry is an absolute
+    # path to the ast file. Let's transform that to a relative path from the
+    # ctu-dir; that is the same format we have in the other cases.
+    if config.extdef_mapping_can_read_pch:
+        path = os.path.relpath(path,
+                               os.path.join(config.ctu_dir,
+                                            triple_arch, 'ast'))
+
     # Normalize path on Windows OS.
-    path = os.path.splitdrive(source_path)[1]
+    path = os.path.splitdrive(path)[1]
     # Make relative path out of absolute.
     path = path[1:] if path[0] == os.sep else path
+
+    # If clang-extdef-mapping can read a pch then the file entry already has
+    # the .ast suffix.
```

Comment:
Is this something worth `assert`in then?

Reply:
No, because there is another case we want to support: when the clang-extdef-mapping cannot read a pch (i.e. older Clang).

Reply:
I don't understand. In this line, we add a the ".ast" suffix for the path, but only when necessary.